temporian/implementation/numpy/operators/tests/assign/assign_test.py

```diff
@@ -0,0 +1,132 @@
+# Copyright 2021 Google LLC.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from absl.testing import absltest
+
+from temporian.implementation.numpy.operators import assign
+from temporian.implementation.numpy.operators.tests.assign.test_data import (
+    different_index,
+)
+from temporian.implementation.numpy.operators.tests.assign.test_data import (
+    with_idx_more_timestamps,
+)
+
+from temporian.implementation.numpy.operators.tests.assign.test_data import (
+    with_idx_same_timestamps,
+)
+
+from temporian.implementation.numpy.operators.tests.assign.test_data import (
+    less_assigned_indexes,
+)
+
+from temporian.implementation.numpy.operators.tests.assign.test_data import (
+    complete_timestamps,
+)
+
+from temporian.implementation.numpy.operators.tests.assign.test_data import (
+    assigned_repeated_timestamps,
+)
+
+from temporian.implementation.numpy.operators.tests.assign.test_data import (
+    assignee_repeated_timestamps,
+)
+
+
+class AssignOperatorTest(absltest.TestCase):
+    def setUp(self) -> None:
+        self.operator = assign.NumpyAssignOperator()
+
+    def test_assigned_repeated_timestamps(self) -> None:
+        self.assertRaisesRegex(
+            ValueError,
+            (
+                "Assigned sequence cannot have repeated timestamps in the same"
+                " index."
+            ),
+            self.operator,
+            assigned_repeated_timestamps.INPUT_1,
+            assigned_repeated_timestamps.INPUT_2,
+        )
+
+    def test_assignee_repeated_timestamps(self) -> None:
+        operator_output = self.operator(
+            assignee_repeated_timestamps.INPUT_1,
+            assignee_repeated_timestamps.INPUT_2,
+        )
+
+        print("Actual:")
```

Comment:
Note that "print" don't work well in tests.
You can either use logging.info (from the absl package; best solution) or do "print(..., flush=True)" (for quick debugging).